### PR TITLE
feat: wing test

### PR DESCRIPTION
**Pull request by eladb**
A simple testing facility for Wing applications:

    wing test ENTRYPOINT.w

Will execute all `cloud.Function`s named `test` or start with `test:` (e.g. `test: bucket is cool`) inside a simulator.

Each test will be executed in sequence inside a separate instance of the simulator, thereby creating an isolated environment between tests (nice!).

A new testing API was added to the `testing.Simulator` class and includes: `listTests()`, `runTest()`, `runAllTests()`.

Here's example output:

Screenshot 2023-01-16 at 11 33 30

*By submitting this pull request, I confirm that my contribution is made under the terms of the 
[Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: apps/wing/src/index.ts [hunk 1]
```diff
@@ -53,6 +53,12 @@ async function main() {
     )
     .action(compile);
 
+  program
+    .command("test")
+    .description("Compiles a Wing program and runs all functions with the word 'test' in their IDs")
```
Comment (eladb) [inline apps/wing/src/index.ts:58]: update to `test` and `test:`

**Comment by ekeren** [comment]
Awesome!!!

Do you mind adding a simple example in the docs?

File: libs/wingsdk/test/testing/simulator.test.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+          },
+        },
+      }
+    );
+
+    new Test(
+      app,
+      "test:bucket2",
+      [
+        "await this.bucket.put('hello', 'world');",
```
Comment (eladb) [inline libs/wingsdk/test/testing/simulator.test.ts:111]: Indent

File: libs/wingsdk/test/testing/simulator.test.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    super(
+      scope,
+      id,
+      Testing.makeHandler(
+        scope,
+        `${id}.handler`,
+        `async handle() { ${code.join("\n")} }`,
+        bindings
+      )
+    );
```
Comment (eladb) [inline libs/wingsdk/test/testing/simulator.test.ts:150]: Indent

**Review by Chriscbr — CHANGES_REQUESTED** [review]
E2E test snapshots are missing

**Review by Chriscbr — APPROVED** [review]

File: libs/wingsdk/src/testing/simulator.ts [hunk 4]
```diff
@@ -386,6 +411,74 @@ export class Simulator {
     this._traceSubscribers.push(subscriber);
   }
 
+  /**
+   * Lists all resource with identifier "test" or that start with "test:*".
+   * @returns A list of resource paths
+   */
```
[the remark this answers is not in the document]
Comment (eladb) [reply libs/wingsdk/src/testing/simulator.ts:417]: Perhaps...

**Pull request merged**